joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -270,7 +276,72 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
       RubyOperators.hashInitializer,
       DispatchTypes.STATIC_DISPATCH
     )
-    callAst(call, argumentAsts)
+    scope.popScope()
+    blockAst(block, List(callAst(call, argumentAsts)))
+  }
+
+  protected def astForAssociationHash(node: Association, tmp: String): List[Ast] = {
+    node.key match {
+      case rangeExpr: RangeExpression =>
+        val expandedList = generateStaticLiteralsForRange(rangeExpr).map { x =>
+          {
```

Comment:
You can now get rid of this brace, since the outer one effectively does this, e.g.
```scala
 generateStaticLiteralsForRange(rangeExpr).map { x =>
            astForSingleKeyValue(x, node.value, tmp)
}
```
Also, the braces are generally only there for multi-line lambdas. Parenthesis are there for single line ones. Another example is that this can be `generateStaticLiteralsForRange(rangeExpr).map(astForSingleKeyValue(_, node.value, tmp))`, but not worth a change in code, just an FYI